### ISSUE TITLE
Failure running “Citation Based on Current Word…” on Sierra

### ISSUE THREAD

**Issue by kociak** [textmate/latex.tmbundle#158]
Dear

I switch recently to textmate 2, with my new mac book pro and Sierra.
When trying to use the beautiful citation autocompletion, I got a very long error message (attached at the end), basically stating that there is a failure  running “Citation Based on Current Word…”. It begins with the following error "in `sort_by': undefined method `<=>' for nil:NilClass (NoMethodError)"
I have no other issues with other completions, even "based on current word"
I have tried to reinstall textmate, then your bundle, nothing changed. The .bib file is managed through jabref, and I never ever had this issue with textmate 1 on my former, macbook air.
Any help appreciated - I can send my text and bib file, but basically it is the third document displaying the same behaviour.
thanks
********
error message:

/Users/mathieukociak/Library/Application Support/TextMate/Managed/Bundles/latex.tmbundle/Support/lib/Ruby/latex.rb:219:in `sort_by': undefined method `<=>' for nil:NilClass (NoMethodError)
	from /Users/mathieukociak/Library/Application Support/TextMate/Managed/Bundles/latex.tmbundle/Support/lib/Ruby/latex.rb:219:in `citations'
	from /Users/mathieukociak/Library/Application Support/TextMate/Managed/Bundles/latex.tmbundle/Support/lib/Ruby/command.rb:304:in `citations'
	from /Users/mathieukociak/Library/Application Support/TextMate/Managed/Bundles/latex.tmbundle/Support/lib/Ruby/command.rb:315:in `insert_citation'
	from Citation Based on Current Word…:9

**Comment by sanssecours** [comment]
Hi kociak,

thank you for the bug report. Could you please add [steps to reproduce the problem](https://github.com/textmate/textmate/wiki/Writing-Bug-Reports)? I tried the following on my machine, which worked:

1. Create a new Document (<kbd>⌘</kbd> + <kbd>N</kbd>)
2. Paste the following text into the document

    ```bib
    @misc{ Deltron3030,
           author = "Deltron 3030",
           title = "Battlesong",
           year = "3030" }
    ```

3. Save the file as `references.bib` in `~/Downloads`
4. Create a new Document (<kbd>⌘</kbd> + <kbd>N</kbd>)
5. Paste the following text:

    ```tex
    \documentclass{article}
    \usepackage{cite}

    \begin{document}

    Del

    \bibliographystyle{plain}
    \bibliography{references}{}
    \end{document}
    ```
6. Save the file as `test.tex` in `~/Downloads`
7. Move the caret to the position right after the word `Del`
8. Invoke ”Citation Based on Current Word” (<kbd>⌥</kbd> + <kbd>⌘</kbd> followed by <kbd>2</kbd>)
9. The document `test.tex` now contains the following text:

    ```tex
    \documentclass{article}
    \usepackage{cite}

    \begin{document}

    \cite{Deltron3030}

    \bibliographystyle{plain}
    \bibliography{references}{}
    \end{document}
    ```

**Comment by kociak** [comment]
Hi René

Thanks. Your test just work perfect. In addition, I tested an alternative to 7-8 :
7-alt Embrace the word `del` with `\cite{}`
8-alt Invoke "Command completion"
leading to the same result, as expected.

Therefore, the problem came from my bib file. After testing different options, I realized that I had entries without keys. Adding keys to that entries solved the problem.
Although it is not good practice to have empty keys, should I report a bug?
Anyway, for me the issue is fixed and I warmly thanks you for the quick help!

**Comment by sanssecours** [comment]
> Therefore, the problem came from my bib file. After testing different options, I realized that I had entries without keys. Adding keys to that entries solved the problem.
> Although it is not good practice to have empty keys, should I report a bug?

While I think it is not ideal to allow empty keys, I can see the appeal in allowing such entries. I fixed the problem in commit 4f20bf6e. Now the [steps from my first comment](https://github.com/textmate/latex.tmbundle/issues/158#issuecomment-271385102) should work, even if the bibliography file `references.bib` looks like this:

```bib
@misc{ Deltron3030,
       author = "Deltron 3030",
       title = "Battlesong",
       year = "3030" }

@misc { , 
       author = "The Cranberries",
       title   = "Empty" }
```

. Thanks again for the bug report.

**Comment by kociak** [comment]
That is pretty nice. Thanks for the hard work making this so nice latex bundle!
